R/create_task_makefile.R

```diff
@@ -82,23 +83,43 @@ create_task_makefile <- function(
   }), use.names=FALSE)
   job_output <- if(isTRUE(ind_complete) || is.na(ind_complete)) {
     if(is.null(ind_dir)) stop('ind_dir must not be NULL when ind_complete=TRUE')
-    file.path(ind_dir, as_ind_file(job_target, ind_ext))
+    file.path(ind_dir, as_ind_file(target_name, ind_ext))
   } else {
-    sprintf('%s_%s', job_target, ind_ext)
+    sprintf('%s_%s', target_name, ind_ext)
   }
+  
+  # prepare the task list for rendering
+  tasks <- unname(task_plan) # remove list element names where they'd interfere with whisker.render
+  for(task in seq_along(tasks)) {
+    # remove lmore ist element names where they'd interfere with whisker.render
+    tasks[[task]]$steps <- unname(tasks[[task]]$steps)
+    for(step in seq_along(tasks[[task]]$steps)) {
+      # add a logical for has_depends
+      tasks[[task]]$steps[[step]]$has_depends <- length(tasks[[task]]$steps[[step]]$depends) > 0
+    }
+  }
+  
   job <- list(
-    target_name = job_target,
+    target_name = target_name,
     # even though target_name is an object (not file), job_command should write
     # to ind_file - again so the calling remake file can use
     # ind_file as its target
     command = {
+      if(!is.null(finalize_fun)){
+        if(isTRUE(ind_complete) | is.na(ind_complete)) {
+          message('ind_complete=', ind_complete, ' is incompatable with ', finalize_fun, '(). Using FALSE')
+          ind_complete <- FALSE
+          # and will need to skip hashing...
+        }
+        sprintf("%s(target_name = %s,\n)", finalize_fun, target_name) # and more w/ ...
```

Comment:
I'm wondering - do we want to make it possible to run the combiner before all the tasks are complete? This is often useful while developing code (e.g., try merging a week of .pngs into a .gif before building the whole year, or work out the code for combining .netcdf files before downloading all the .netcdf data). There are a few ways a user could make that happen without modifying this code, but they're each a little bit clunky, and there might be a clever way to implement this combiner so that it only operates on the completed steps, whichever those are...food for thought.

Reply:
Not sure - my initial response would be that they should make the dev version of the makefile a truncated number of tasks. 

or, you could edit your combiner in dev mode:
```r
my_combine <- function(target_name, ...){
   files <- c(...)[1:10] 
   message("I am skipping the rest of the files!!!")
   # do things
   write_feather(target_name, stuff)
}
```

Reply:
Editing the combiner wouldn't work because remake would see all the ...s and wouldn't run the combiner until they'd all been built.

So far today I agree with your initial response. I think I've had reasons not to want to do it that way in the past, but I can't dredge them up today at all.

Reply:
ahh yep, you are right.